lib/ae_mdw/sync/server.ex

```diff
@@ -108,8 +120,23 @@ defmodule AeMdw.Sync.Server do
   defp process_state(s), do: s
 
   defp spawn_sync(from_height, to_height) do
+    from_mbi =
+      case Block.last_synced_mbi(from_height) do
+        {:ok, mbi} -> mbi + 1
+        :none -> -1
+      end
+
+    block_index = {from_height, from_mbi}
+    db_state = State.new()
+    from_txi = Block.next_txi()
+
     spawn_link(fn ->
-      Block.sync(from_height, to_height)
+      {blocks_mutations, _next_txi} = Block.blocks_mutations(block_index, from_txi, to_height)
+
+      _new_state =
+        Enum.reduce(blocks_mutations, db_state, fn {block_index, block, mutations}, state ->
+          commit_mutations(state, block_index, block, mutations)
+        end)
```

Comment:
There are some blocks that take a long time to load them from node DB and dry-run the events. What you think of having commits for every height?

Reply:
That's a good idea, I'd like to add that on a following PR so we can test it thoroughly from scratch